internal/metrics/listener.go

```diff
@@ -49,5 +53,47 @@ func runServer(port string, registry prometheus.Gatherer) {
 	})
 	address := fmt.Sprintf(":%v", port)
 	glog.Infof("Starting Prometheus listener on: %v%v", address, metricsEndpoint)
-	glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServe(address, nil))
+	if prometheusSecret == nil {
+		glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServe(address, nil))
+	} else {
+		// Unfortunately, http.ListenAndServeTLS() takes a filename instead of cert/key data, so we
+		// Write the cert and key to a temporary file. We create a unique file name to prevent collisions.
+		certFileName := fmt.Sprintf("%v-%s", time.Now().Unix(), "nginx-prometheus.cert")
+		keyFileName := fmt.Sprintf("%v-%s", time.Now().Unix(), "nginx-prometheus.key")
+		certFile, err := writeTempFile(prometheusSecret.Data[api_v1.TLSCertKey], certFileName)
+		if err != nil {
+			glog.Errorf("failed to create cert file for prometheus: %w", err)
+		}
+		defer removeTemporaryFile(certFile)
```

Comment:
not sure why we need to remove those files. If the IC process terminates or exists, Docker runtime will remove the IC container along with all its files.

Reply:
I'd rather not leave certs hanging around on the file system unless they are needed again.

Reply:
it is a good practice! 

however, note that `removeTemporaryFile()` will never be executed.  That's because ` http.ListenAndServeTLS(address, certFile.Name(), keyFile.Name(), nil)` blocks forever. Even if it doesn't block, but returns an error, glog.Fatal will print it and terminate the IC process, so that `removeTemporaryFile()` will not be called.

 If the intention is to remove the files immediately after `http.ListenAndServeTLS` is called , then the current approach doesn't work :(